### DISCUSSION TITLE
Is there any way to use HyperlinkedModelSerializer without having to define the url view_name?

### DISCUSSION THREAD

**Discussion by brunodantas** [encode/django-rest-framework#7946] [Question & Answer]
This is basically a duplicate of a [stackoverflow question](https://stackoverflow.com/q/58789733/2853953) that had no answer.

When I implement a [`HyperlinkedModelSerializer`](https://www.django-rest-framework.org/api-guide/serializers/#hyperlinkedmodelserializer) subclass without redefining the url, I get the following error.

> django.core.exceptions.ImproperlyConfigured: Could not resolve URL for hyperlinked relationship using view name "client-detail". You may have failed to include the related model in your API, or incorrectly configured the `lookup_field` attribute on this field.

It only works if I manually set the url attribute as follows.

`url = HyperlinkedIdentityField(view_name='myapp:myview-detail')`

Is there any way to do this without having to set the url attribute for every serializer class?

**Comment by jerinpetergeorge** [comment]
> Is there any way to do this without having to set the url attribute for every serializer class?

If you're not setting the `url` parameter, how would be the result? What is your expected behavior?

**Reply by brunodantas** [reply]
Current behavior: it is required to set the url attribute manually for every serializer class in order to specify the app name. Otherwise, I get that `ImproperlyConfigured` error.

Example:

```
class Example1Serializer(HyperlinkedModelSerializer):
    url = HyperlinkedIdentityField(view_name='myapp:example1-detail') 
    class Meta:
    ...

class Example2Serializer(HyperlinkedModelSerializer):
    url = HyperlinkedIdentityField(view_name='myapp:example2-detail') 
    class Meta:
    ...

class Example3Serializer(HyperlinkedModelSerializer):
    url = HyperlinkedIdentityField(view_name='myapp:example3-detail') 
    class Meta:
    ...
```

Desired behavior: some way to specify the app name only once instead of setting the url parameter manually for every class. Example:

```
APP_NAME = "myapp"

class Example1Serializer(HyperlinkedModelSerializer):
    class Meta:
    ...

class Example2Serializer(HyperlinkedModelSerializer):
    class Meta:
    ...

class Example3Serializer(HyperlinkedModelSerializer):
    class Meta:
    ...
```
